patches/server/0814-Overridable-YAML-configuration.patch

```diff
@@ -0,0 +1,330 @@
++        if (override != null) {
++            return override;
++        }
++        return super.getString(path, def);
++    }
++
++    @Override
++    public int getInt(@NotNull String path, int def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Integer.parseInt(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getInt(path, def);
++    }
++
++    @Override
++    public boolean getBoolean(@NotNull String path, boolean def) {
++        String override = getOverride(path);
++        if (override != null) {
++            return Boolean.parseBoolean(override);
++        }
++        return super.getBoolean(path, def);
++    }
++
++    @Override
++    public double getDouble(@NotNull String path, double def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Double.parseDouble(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getDouble(path, def);
++    }
++
++    @Override
++    public long getLong(@NotNull String path, long def) {
++        String override = getOverride(path);
++        if (override != null) {
++            try {
++                return Long.parseLong(override);
++            } catch (NumberFormatException ignored) {
++
++            }
++        }
++        return super.getLong(path, def);
++    }
++
++    private @Nullable String getOverride(@NotNull String path) {
++        for (OverrideSource source : sources) {
++            String override = source.getOverride(path, group);
++            if (override != null) {
++                return override;
++            }
++        }
++        return null;
++    }
++
++    /**
++     * Creates a new {@link OverridableYamlConfiguration}, loading from the given file.
++     * <p>
++     * Any errors loading the Configuration will be logged and then ignored.
++     * If the specified input is not a valid config, a blank config will be
++     * returned.
++     * <p>
++     * The encoding used may follow the system dependent default.
++     *
++     * @param file Input file
++     * @param group Configuration group name
++     * @return Resulting configuration
++     * @throws IllegalArgumentException Thrown if file is null
++     */
++    @NotNull
++    public static OverridableYamlConfiguration loadConfiguration(@NotNull File file, @NotNull String group) {
++        Validate.notNull(file, "File cannot be null");
++
++        OverridableYamlConfiguration config = new OverridableYamlConfiguration(group);
++
++        try {
++            config.load(file);
++        } catch (FileNotFoundException ex) {
++        } catch (IOException ex) {
++            Bukkit.getLogger().log(Level.SEVERE, "Cannot load " + file, ex);
++        } catch (InvalidConfigurationException ex) {
```

Comment:
Could really just collapse these exceptions, no need to stick to the spigot double logic.